db/funds.go

```diff
@@ -81,9 +80,27 @@ func (f *FundsDB) InsertLog(ctx context.Context, logs ...*FundsLog) error {
 	return nil
 }
 
-func (f *FundsDB) Logs(ctx context.Context) ([]FundsLog, error) {
+func (f *FundsDB) Logs(ctx context.Context, first *time.Time, offset int, limit int) ([]FundsLog, error) {
```

Comment:
Nit: Maybe it's just me, but I don't associate `first` with time. I suggest renaming this to `cursor`.